### PR TITLE
create calculator

### PR DESCRIPTION
**Pull request by DiegoCuevas**
Team : the calculator

Christopher Roa
Lian Nivin
Diego Cuevas

### CODE REVIEW

**Review by valevassallo — COMMENTED** [review]
1. No hay "operator precedence"
_There's no operator precedence_
2. No permite calcular con paréntesis
_It doesn't let us calculate with parenthesis_
3. No permite calcular si hay espacios
_It doesn't let us calculate if there are spaces_

File: week-1/calculator.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            when "+"
+                res=num[i-1].to_i.send :+,num[i+1].to_i
+            when "-"
+                res=num[i-1].to_i.send :-,num[i+1].to_i                       
+            else
+                res=0;
+        end
+        num[i-1]=res
+        num.delete_at(i)
+        num.delete_at(i)
```
Comment (valevassallo) [inline week-1/calculator.rb:24]: Línea duplicada
_Duplicated line_

File: week-1/calculator.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+puts "ingrese su operacion:"
+op = gets.chomp
+num = op.split(/(\d+\.?\d*)/).reject(&:empty?)
+i=0
+signos = ["+","-","*","/"]
+while i < num.length
+    if signos.include? (num[i])
+        case num[i]
+           when "**"
+                res=num[i-1].to_i.send :**,num[i+1].to_i
```
Comment (KattyaCuevas) [inline week-1/calculator.rb:10]: El método `send` permite strings, esto se puede mejorar.

File: week-1/calculator.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+puts "ingrese su operacion:"
+op = gets.chomp
+num = op.split(/(\d+\.?\d*)/).reject(&:empty?)
+i=0
+signos = ["+","-","*","/"]
```
Comment (KattyaCuevas) [inline week-1/calculator.rb:5]: Espacios faltantes

File: week-1/calculator.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+puts "ingrese su operacion:"
+op = gets.chomp
+num = op.split(/(\d+\.?\d*)/).reject(&:empty?)
```
Comment (KattyaCuevas) [inline week-1/calculator.rb:3]: Qué hace `/(\d+\.?\d*)/` ?

Comment (KattyaCuevas) [reply week-1/calculator.rb:3]: Qué significa `&:empty?`?

File: week-1/calculator.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                res=num[i-1].to_i.send :-,num[i+1].to_i                       
+            else
+                res=0;
+        end
+        num[i-1]=res
+        num.delete_at(i)
+        num.delete_at(i)
+        i = i - 1 
+    end
+i = i+1
```
Comment (KattyaCuevas) [inline week-1/calculator.rb:27]: Faltan espacios

File: week-1/calculator.rb (hunk 1 above)
Comment (DiegoCuevas) [reply week-1/calculator.rb:24]: tendriamos que eliminar el i y i+1 
pero al eliminar i automaticamente el elemento de indice i+1 se vuelve i 
esta duplicada por necesidad 
we had to eliminate the "i" and "i +1" but when we eliminate "i" automatically the index element "i+1" becomes "i", this duplicated was necessary

**Comment by DiegoCuevas** [comment]
@valevassallo  
- No hay "operator precedence"
no pudimos resolverlo a tiempo lo siento
- No permite calcular con paréntesis 
se indico que era opcional
- No permite calcular si hay espacios 
si no permite porque al hacer un espacio lo toma como un elemento en el array

File: week-1/calculator.rb (hunk 4 above)
Comment (DiegoCuevas) [reply week-1/calculator.rb:3]: queria hacer un split con multiples opciones

File: week-1/calculator.rb (hunk 5 above)
Comment (DiegoCuevas) [reply week-1/calculator.rb:27]: no entiendo (?)

File: week-1/calculator.rb (hunk 2 above)
Comment (DiegoCuevas) [reply week-1/calculator.rb:10]: si trataremos de arreglarlo

File: week-1/calculator.rb (hunk 1 above)
Comment (arturopuente) [reply week-1/calculator.rb:24]: Quizá en vez de un case, mandarle directamente el signo!:
```ruby
sign = num[i]
number = num[i - 1].to_i
other_number = num[i + 1].to_i
res = number.send(sign.to_sym, other_number)
```

File: week-1/calculator.rb (hunk 5 above)
Comment (arturopuente) [reply week-1/calculator.rb:27]: `i = i+1` => `i = i + 1`

File: week-1/calculator.rb (hunk 1 above)
Comment (luciagirasoles) [reply week-1/calculator.rb:24]: (*english lesson about giving a code review in english) Why is line 23 duplicated? Can you give an explanation about it, please?

**Comment by luciagirasoles** [comment]
(*English lesson about giving a code review in English)  When we tested this code, we found:
1. Addition and subtraction were executed first than multiplication and division.
2. Parenthesis are not allowed to group operations.
3. If spaces are included, it doesn't work.

**Pull request merged**